sub-packages/bionemo-esm2/src/bionemo/esm2/data/datamodule.py

```diff
@@ -172,11 +171,13 @@ def setup(self, stage: str = "") -> None:
             hasattr(self, "trainer") and self.trainer is not None
         ), "Setup should be completed when trainer and config are attached."
 
-    def _create_dataloader(self, dataset, **kwargs) -> torch.utils.data.DataLoader:
+    def _create_dataloader(self, dataset, **kwargs) -> WrappedDataLoader:
         assert self._tokenizer.pad_token_id is not None, "Tokenizer must have a pad token id."
+        self.init_global_step = self.trainer.global_step
+        self.data_sampler.init_global_step = self.init_global_step
 
-        return torch.utils.data.DataLoader(
-            dataset,
+        return WrappedDataLoader(
+            dataset=dataset,
```

Comment:
You forgot to pass in `mode`

Reply:
OOOOOH I see it sneaks in through `kwargs`. I have a pretty strong readability preference to expose this and pass it through since it's so important.

Reply:
So just add like a `mode:Literal[whatever]` arg and then pass `mode` into the `WrappedDataLoader` along with a comment saying "mode is used to determine if consumed_samples in the sampler should be initialized to 0 (non-train loaders) or be set to the previous consumed samples in the case of checkpoint resumption.` or something. I think over-communicating this is important for users who copy/paste code to see for when they modify.